src/libcore/slice/rotate.rs

```diff
@@ -88,8 +85,8 @@ pub unsafe fn ptr_rotate<T>(mut left: usize, mid: *mut T, mut right: usize) {
         }
     }
 
-    let rawarray = RawArray::new();
-    let buf = rawarray.ptr();
+    let rawarray = MaybeUninit::<RawArray<T>>::uninitialized();
+    let buf = rawarray.get_ref().ptr();
```

Comment:
Isn't this also not legitimate code? `get_ref` is only defined as far as I can tell on initialized memory, but it's not initialized here.